File: Padroes de Projeto/Inversao de Controle/Java/Usuario.java
public class Usuario {
    private String nome;
    private String telefone;
    private String email;

    private Usuario(String nome, String telefone, String email) {
        this.nome = nome;
        this.telefone = telefone;
        this.email = email;

        validate();
    }

    public static Usuario criar(String nome, String telefone, String email) {
        return new Usuario(nome, telefone, email);
    }

    public void update(String nome, String email, String telefone) {
        this.nome = nome;
        this.email = email;
        this.telefone = telefone;

        validate();
    }

    private void validate() {
        if (nome.isEmpty() || nome == null) {
            throw new IllegalArgumentException("Nome é obrigatório!");
        }

        if (email.isEmpty() || email == null) {
            throw new IllegalArgumentException("Email é obrigatório!");
        }

        if (telefone.isEmpty() || telefone == null) {
            throw new IllegalArgumentException("Telefone é obrigatório!");
        }
    }

    public String email() {
        return email;
    }

    public String nome() {
        return nome;
    }

    public String telefone() {
        return telefone;
    }
}

File: Padroes de Projeto/Inversao de Controle/Python/Usuario.py
from wsgiref import validate


class Usuario:
    def __init__(self, nome, email, telefone):
        self.nome = nome
        self.email = email
        self.telefone = telefone

        self.validated()
    
    def criar(nome, email, telefone):
        return Usuario(nome, email, telefone)
    
    def validated(self):
        if self.nome is None or not self.nome:
            raise Exception("Nome é obrigatório!")
        if self.email is None or not self.email:
            raise Exception("Email é obrigatório!")
        if self.telefone is None or not self.telefone:
            raise Exception("Telefone é obrigatório!")
    
    def update(self, nome, email, telefone):
        self.nome = nome
        self.email = email
        self.telefone = telefone

        self.validated()
    
    def nome(self):
        return self.nome

    def email(self):
        return self.email

    def telefone(self):
        return self.telefone
    